### PR TITLE
Triggers websocket when a folder is recursively created

### PR DESCRIPTION
**Pull request by jaxoncreed**
Solves https://github.com/solid/node-solid-server/issues/1497

### CODE REVIEW

File: lib/ldp.js [hunk 1]
```diff
@@ -229,8 +229,10 @@ class LDP {
 
     // First check if we are above quota
     let isOverQuota
+    // For some reason someone thought it was a bright idea to make url actually a
+    // req sometimes but not all the time 🙄. So Now I have to account for that
```
Comment (michielbdejong) [inline lib/ldp.js:233]: Can you specify who "I" is, or just rewrite this so it's more "us" and "we" or "the code needs to ..."

**Review by michielbdejong — APPROVED** [review]
Can you also unskip the test for this in the nss-skips branch of solid-crud-tests please?

File: lib/handlers/patch.js [hunk 2]
```diff
@@ -60,7 +62,19 @@ async function patchHandler (req, res, next) {
     const pathSplit = path.split('/')
     const directory = pathSplit.slice(0, pathSplit.length - 1).join('/')
     if (!fs.existsSync(directory)) {
-      fx.mkdirSync(directory, { recursive: true })
+      const firstDirectoryRecursivelyCreated = await promisify(mkdirp)(directory)
```
Comment (timbl) [inline lib/handlers/patch.js:65]: The mkdirp function https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md doesn't seem to return the directory, only an error code. Are you below  in 69 assuming it firstDirectoryRecursivelyCreated will be the directory name?

Comment (jaxoncreed) [reply lib/handlers/patch.js:65]: I removed fs extra from node solid server as this seemed to be the only place it was used. I replaced it with https://www.npmjs.com/package/mkdirp which does return the directory.

File: lib/ldp.js (hunk 1 above)
Comment (TallTed) [inline lib/ldp.js:233]:
```suggestion
    // Someone had a reason to make url actually a req sometimes but not
    // all the time. So now we have to account for that, as done below.
```

**Pull request merged**